dev/domain-controller/new-admin-vm/Install-CAM.ps1

```diff
@@ -1320,7 +1330,159 @@ brokerLocale=en_US
 
 				Restart-Service $using:brokerServiceName
             }
-        }
+		}
+		
+		Script RegisterCam
+		{
+			DependsOn  = @("[Script]Install_Auth_file")
+			GetScript  = { @{ Result = "RegisterCam" } }
+
+            TestScript = { 
+				if ( $env:CAM_USERNAME -and $env:CAM_PASSWORD -and $env:CAM_TENANTID -and $env:CAM_URI -and $env:CAM_DEPLOYMENTID)
+				{
+					return $true
+				} else {
+					return $false
+				}
+			}
+
+            SetScript  = {
+				##
+				# Do this so SSL Errors are ignored
+				add-type @"
+	using System.Net;
+	using System.Security.Cryptography.X509Certificates;
+	public class TrustAllCertsPolicy : ICertificatePolicy {
+		public bool CheckValidationResult(
+			ServicePoint srvPoint, X509Certificate certificate,
+			WebRequest request, int certificateProblem) {
+			return true;
+		}
+	}
+"@
+				[System.Net.ServicePointManager]::CertificatePolicy = New-Object TrustAllCertsPolicy
+				##
+
+				# Read in Authorization Information
+				Get-Content "$env:AZURE_AUTH_LOCATION" | Foreach-Object{
+					$var = $_.Split('=', 2)
+					New-Variable -Name $var[0] -Value $var[1]
+				}
+
+				$camSaasBaseUri = $using:camSaasUri
+				$camRegistrationError = ""
+				for($idx = 0; $idx -lt $using:retryCount; $idx++) {
+					try {
+						$userRequest = @{
+							username = $client
+							password = $key
+							tenantId = $tenant
+						}
+						$registerUserResult = ""
+						try {
+							$registerUserResult = Invoke-RestMethod -Method Post -Uri ($camSaasBaseUri + "/api/v1/auth/users") -Body $userRequest
+						} catch {
+							$registerUserResult = ConvertFrom-Json $_.ErrorDetails.Message
+						}
+						Write-Verbose (ConvertTo-Json $registerUserResult)
+						# Check if registration succeeded or if it has been registered previously
+						if( !(($registerUserResult.code -eq 201) -or ($registerUserResult.data.reason.ToLower().Contains("already exist"))) ) {
+							throw ("Failed to register with CAM. Result was: " + (ConvertTo-Json $registerUserResult))
+						}
+
+						[System.Environment]::SetEnvironmentVariable("CAM_USERNAME", $userRequest.username, "Machine")
```

Comment:
Much nicer than the registry setting I was doing. So if you set 'machine' it doesn't change for the running process but will change for any new processes (and after reboot, since it's setting the registry)?